pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -528,6 +536,16 @@ func (s *Controller) WorkloadInstanceHandler(wi *model.WorkloadInstance, event m
 	s.mutex.Unlock()
 
 	s.edsUpdate(instances)
+
+	// ServiceEntry with WorkloadEntry results in STRICT_DNS cluster with hardcoded endpoints
+	// need to update CDS to refresh endpoints
+	// https://github.com/istio/istio/issues/39505
+	pushReq := &model.PushRequest{
+		Full:           true,
```

Comment:
Doesn't this mean we are triggering a full push every time _any_ WorkloadInstance changes? That seems like it would be quite bad for performance?

Reply:
And only for the DNS edge case - for typical usage it provides no benefit, only cost. If I understand right?

Reply:
I was wondering how best to narrow the solution down.  Would setting a flag while looping over the ServiceEntries for configs with `resolution: DNS` and `workloadSelector != nil` cut back on the impact? https://github.com/istio/istio/blob/8705c5417fb70a3790944186b3605da687f8d02c/pilot/pkg/serviceregistry/serviceentry/controller.go#L498

Reply:
@howardjohn I deployed some other apps and ServiceEntries of type DNS and with the latest changes - the full push is only triggered 3 times in the example from the linked issue: 1) pod create, 2) updated pod created and 3) original pod is deleted.  Are there any other false positive scenarios you are you can think of?